src/Microsoft.VisualStudio.Services.Agent/ConfigurationStore.cs

```diff
@@ -24,6 +24,9 @@ public sealed class AgentSettings
         [IgnoreDataMember]
         public bool IsHosted => !string.IsNullOrEmpty(NotificationPipeName) || !string.IsNullOrEmpty(NotificationSocketAddress);
 
+        [DataMember(EmitDefaultValue = false)]
+        public string Fingerprint { get; set; }
```

Comment:
I thought this would end up being an array of strings.

Reply:
I didn't want to expose that until we are sure it will work end to end and since it wasn't part of the requirements I kept it singular for now. Technically you could pass a comma-separated list of public keys to this property and it _should_ work. But not officially supported at the moment.